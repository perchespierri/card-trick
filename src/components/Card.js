import React from 'react';
import '../styles/Card.css';

export default function Card({ card: {value, suit} }) { 
  return (
      <li
        key={`${suit} ${value}`}
        className={`Card Card--${suit}`}
      >
        {value}
        <div className={`Card__suit--image Card__suit--${suit}`} />
      </li>
  )
}
