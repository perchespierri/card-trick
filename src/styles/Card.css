.Stack .Card {
  border: solid 2px rgb(0, 0, 0);
  display: inline-block;
  border-radius: 10px;
  width: 100px;
  height: 150px;
  background-color: rgb(255, 255, 255);
  padding: 3px;
  margin: 5px;
}

.Card {
  font-size:25px;
  font-weight: bold;
}

.Card--diamonds,
.Card--hearts {
  color : red;
}

.Card .Card__suit--image {
	background-image: url('../images/sprites.png');
  height: 100px;
	width: 90px;
  margin-left: 8px;
  margin-top: 5px;
}

.Card .Card__suit--diamonds {
	background-position-y: 100px;
}

.Card .Card__suit--hearts {
	background-position-x: 90px;
}

.Card .Card__suit--clubs {
	background-position-x:90px;
	background-position-y:100px;
}
