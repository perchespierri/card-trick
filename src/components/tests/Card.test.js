import { render } from '@testing-library/react'
import Card from '../Card';

describe('Card component', () => {

  it.each(['hearts', 'spades', 'diamonds', 'clubs'])
    (`Renders correctly when suit is %s`, (suit) => {    
      const testCard = { value: 'A', suit: suit };
      const { container } = render(<Card card={ testCard } />);
      
      expect(container).toMatchSnapshot();
    });

    it('Renders card of any value correctly', () => {
      const testCard = { value: 'TEST VALUE', suit: 'clubs' };
      const { container } = render(<Card card={ testCard } />);
      
      expect(container).toMatchSnapshot();
    });
})
