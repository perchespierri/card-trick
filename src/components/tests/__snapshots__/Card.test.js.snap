// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component Renders card clubs correctly 1`] = `
<div>
  A
  <div
    class="Card__suit--image Card__suit--clubs"
  />
</div>
`;

exports[`Card component Renders card of any value correctly 1`] = `
<div>
  TEST VALUE
  <div
    class="Card__suit--image Card__suit--clubs"
  />
</div>
`;

exports[`Card component Renders card of diamonds correctly 1`] = `
<div>
  A
  <div
    class="Card__suit--image Card__suit--diamonds"
  />
</div>
`;

exports[`Card component Renders card of hearts correctly 1`] = `
<div>
  A
  <div
    class="Card__suit--image Card__suit--hearts"
  />
</div>
`;

exports[`Card component Renders card of spades correctly 1`] = `
<div>
  A
  <div
    class="Card__suit--image Card__suit--spades"
  />
</div>
`;
